### ISSUE TITLE
Mxnet produce 'nan' training error!

### ISSUE THREAD

**Issue by liuzhi136** [apache/incubator-mxnet#8096]
I build my own model by mx.sym interface. I then starts the training progress. It performs well at the beginning epochs. It then produce the 'nan' training and validation errors. Does any encounter this kind of problem? I'm looking forward any explanation or solution. Any help about will be appreciated!
The training traces looks like below:
![1593586835](https://user-images.githubusercontent.com/13534043/31003386-01f6435a-a522-11e7-9ac1-cbf69d206aca.jpg)

**Comment by solin319** [comment]
May be gradient explosion.
Try to use relu as an activation function or decrease learning rate.

**Comment by eric-haibin-lin** [comment]
specify `clip_gradient` in the optimizer also helps sometimes

**Comment by liuzhi136** [comment]
How to choose the value of clip_gradient? Any common used advices over parameter clip_gradient? @eric-haibin-lin

**Comment by Ishitori** [comment]
It is a process of trial and error to chose good gradient clipping value. Maybe you can try to follow this advice: https://www.quora.com/How-should-I-set-the-gradient-clipping-value

**Comment by ThomasDelteil** [comment]
@szha can we close this PR? Requester has been inactive for more than 8 months and issue contains already good suggestions.

**Comment by eric-haibin-lin** [comment]
https://github.com/apache/incubator-mxnet/pull/11795 might fix it if you have embedding in your network
